nix/docker.nix

```diff
@@ -27,18 +27,22 @@ let
   defaultPort = "8090";
   dataDir = "/data";
 
-  startScript = writeScriptBin "start-cardano-wallet-${backend}" ''
-    #!${runtimeShell}
-    set -euo pipefail
+  startScript =
+    let
+      suffix = if backend == "shelley" then "" else "-" + backend;
+      walletExeName = "cardano-wallet" + suffix;
+    in writeScriptBin "start-cardano-wallet-${backend}" ''
+      #!${runtimeShell}
+      set -euo pipefail
 
-    # set up data volume
-    export XDG_DATA_HOME=/
-    mkdir -p ${dataDir}
-    ln -s ${dataDir} /cardano-wallet
+      # set up data volume
+      export XDG_DATA_HOME=/
+      mkdir -p ${dataDir}
+      ln -s ${dataDir} /cardano-wallet
 
-    export LOCALE_ARCHIVE="${glibcLocales}/lib/locale/locale-archive"
-    exec ${exe}/bin/cardano-wallet-${backend} "$@"
-  '';
+      export LOCALE_ARCHIVE="${glibcLocales}/lib/locale/locale-archive"
+      exec ${exe}/bin/${walletExeName} "$@"
```

Comment:
Since it's a bash script, you could "cheat" and use a glob.
```
exec ${exe}/bin/cardano-wallet* "$@"
```

Reply:
On second thought, that would break if we for some reason would end up with multiple matches?

If the cost is the two lines:
```nix
  suffix = if backend == "shelley" then "" else "-" + backend;
  walletExeName = "cardano-wallet" + suffix;
```
then maybe being explicit is worth it.

Reply:
The glob will be safe to use because this is the bin directory within a backend-specific package. But explicit filenames are probably easier to maintain.